stargate/crates/stargate-lib/src/lib.rs

```diff
@@ -0,0 +1,71 @@
+use crate::request_pipeline::executor::execute_query_plan;
+use crate::request_pipeline::service_definition::ServiceDefinition;
+use crate::transports::http::{GraphQLResponse, RequestContext};
+use apollo_query_planner::helpers::directive_args_as_map;
+use apollo_query_planner::{QueryPlanner, QueryPlanningOptionsBuilder};
+use graphql_parser::schema;
+use std::collections::HashMap;
+
+pub mod common;
+mod request_pipeline;
+pub mod transports;
+mod utilities;
+
+#[derive(Clone)]
+pub struct Stargate<'app> {
+    service_list: HashMap<String, ServiceDefinition>,
+    pub planner: QueryPlanner<'app>,
+}
+
+impl<'app> Stargate<'app> {
+    pub fn new(schema: &'app str) -> Stargate<'app> {
+        // TODO(ran) FIXME: gql validation on schema
+        let planner = QueryPlanner::new(schema);
+        let service_list = get_service_list(&planner.schema);
+        Stargate {
+            planner,
+            service_list,
+        }
+    }
```

Comment:
Can `get_service_list` be a fn that lives on a `QueryPlanner` instance? Then `Stargate` would only need a planner passed in to its `new` fn, which could then get the `service_list` itself.

Reply:
It can, I tried that at some point. I think I landed on this because it makes sense for the service list to be a property of Stargate. It felt a little awkward to get that from the planner in terms of ownership.

Reply:
Oh I remember, in order to do that, query-planner (which is a different crate) would need to either be aware of `ServiceDefinition` which comes from stargate-lib, and so that definition would have to move, and that has cascading effects.